lib/galaxy/model/security.py

```diff
@@ -1808,3 +1783,46 @@ def is_foreign_key_violation(error):
         # If this is a PostgreSQL foreign key error, then error.orig is an instance of psycopg2.errors.ForeignKeyViolation
         # and should have an attribute `pgcode` = 23503.
         return int(getattr(error.orig, "pgcode", -1)) == 23503
+
+
+def _get_valid_roles_case1(session, search_query, is_admin, limit, page, page_limit):
```

Comment:
What do you think about this alternative name?

```suggestion
def _get_valid_roles_exposed(session, search_query, is_admin, limit, page, page_limit):
```

Reply:
Much better. I was at a loss for a suitable name!